sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -406,24 +404,24 @@ static gnrc_pktsnip_t *_add_pios(gnrc_ipv6_netif_t *ipv6_iface, gnrc_pktsnip_t *
             continue;
         }
 
-        if (_pio_from_iface_addr(&tmp, &ipv6_iface->addrs[i], pkt)) {
-            if (tmp != NULL) {
-                pkt = tmp;
+        if (_pio_from_iface_addr(res, &ipv6_iface->addrs[i], pkt)) {
+            if (*res != NULL) {
+                pkt = *res;
```

Comment:
(resulting in memory leaks in https://github.com/RIOT-OS/RIOT/pull/4174/files#diff-eb98861cc04f206d3376ec33720dfcc7R413)